File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/body/bike/BookBikeBody.java
package com.lattis.ellipse.data.network.model.body.bike;

import com.google.gson.annotations.SerializedName;

public class BookBikeBody {

    @SerializedName("bike_id")
    private int bike_id;

    @SerializedName("by_scan")
    private Boolean by_scan=null;

    @SerializedName("latitude")
    private Double latitude=null;

    @SerializedName("longitude")
    private Double longitude=null;

    public BookBikeBody(int bike_id,boolean by_scan,double latitude,double longitude){
        this.bike_id=bike_id;

        if(by_scan) {
            this.by_scan = by_scan;
            this.latitude = latitude;
            this.longitude = longitude;
        }
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/entity/body/bike/BookBikeBody.kt
package com.lattis.data.entity.body.bike

import com.google.gson.annotations.SerializedName

class BookBikeBody(@field:SerializedName("bike_id")
                   private val bike_id: Int, by_scan: Boolean, latitude: Double?, longitude: Double?,device_token:String,pricing_option_id:Int?) {

    @SerializedName("by_scan")
    private var by_scan: Boolean? = null

    @SerializedName("latitude")
    private var latitude: Double? = null

    @SerializedName("longitude")
    private var longitude: Double? = null

    @SerializedName("device_token")
    private val device_token: String


    @SerializedName("pricing_option_id")
    private var pricing_option_id: Int?



    init {

        if (by_scan) {
            this.by_scan = by_scan
            this.latitude = latitude
            this.longitude = longitude
        }
        this.pricing_option_id = pricing_option_id
        this.device_token = device_token
    }
}
